src/fec_base.h

```diff
@@ -111,6 +112,10 @@ class FecCode {
     size_t pkt_size; // packet size, i.e. number of words per packet
     size_t buf_size; // packet size in bytes
 
+    // Length of operating codeword. It's calculated by derived Class
+    // FIXME: move n to protected
+    T n;
```

Comment:
I'm not really fond of this kind of thing (we already have it for `inv_power_something`.

Relying on a child class to initialize a parameter of its parents is brittle and kind of break the encapsulation.

A better approach may be to have a abstract method (like `T codeword_len() = 0;`) that returns this info.
I see two benefit to this approach:
- you are sure that the child class will initialize it (i.e define the function) because it's a purely abstract method
- each child can compute how it want and don't need to know that it's parent has a field `n`.